doc/developers/contributing.rst

```diff
@@ -464,7 +454,9 @@ is a corner-stone of the scikit-learn development process. For this
 purpose, we use the `nose <http://nose.readthedocs.io/en/latest/>`_
 package. The tests are functions appropriately named, located in `tests`
 subdirectories, that check the validity of the algorithms and the
-different options of the code.
+different options of the code. 
+
+Regression testing is used to identify and validate any code committed to the repository and see if it does not break any existing code. Any changes made to the existing code in repository such as feature enhancements, bug fixes, document changes, hot fixes, config changes or any other changes to the scikit-learn repository are tested using regression tests. During these tests both functional and non-functional areas of the features are tested. Regression testing also ensures that the change being proposed works well with other parts of the software as a whole in an integrated way. Any bug-fix or change in sklearn repository needs a regression test due to the above reasons.
```

Comment:
I agree. I would go with something simpler like:

```
In addition, a non-regression test should go along each bug-fix to ensure that such
bug will not occur in the future.
```

In addition, try to not go over 80 characters also in `*.rst`.